Function to reindex all segments in a message

jay-knight opened the pull request:
This adds a new function to the Message class to re-index all of the segments in a message. When we're reading and building multiple messages, the index/sequence ids can be added in weird and unpredictable ways because of the static/global `$setId` variables.

This solution lets you add any segments you need and then step through and re-index all of the segments so that each segment types starts with 1 and increases by 1 for each segment of that type.

I think a better solution might be to keep up with the segment ids in the Message class (as an instance variable, instead of static variables on each segment class) so that as you add segments they can be set/incremented if that option is set. And multiple messages wouldn't interfere with each other.

senaranya commented:
Thanks for adding this. Can you please add it in README.md under the section "Handling segments and fields"?

jay-knight commented:
I've added a bit about the new function in README.md.

I also changed the function to only index segments whose class has a `setID()` function. Some other segments from the standard don't have a Set ID, so this should be more future compatible if those segments get implemented in this library. 

Do you think it would be better to have an interface (`HasSetID`?) that classes with this field could implement and check if each class implements that interface:

```php

namespace app/HL7;

interface HasSetID {
    public function setID(int $value, int $position = 1);
    public function getID(int $position = 1);
}
```

```php
class PID extends Segment implements HasSetID {
    [...]
}

```

```php
if ($segment instanceof HasSetID) {
    continue;
}
```

senaranya commented:
Thank you for updating README. I think `method_exists()` is fine for now, it does the job with minimal changes. We can later refactor it to interface if need arises

shadowhand commented:
> Do you think it would be better to have an interface (`HasSetID`?) that classes with this field could implement and check if each class implements that interface

The interface `HasSetId` would be a great name for a trait that implements the interface, but not a good name for an interface. I would suggest `SetIdentifier` or `Identified`:

```php
if ($segment instance Identified) { /* ... */ }
```
